src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs

```diff
@@ -0,0 +1,237 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.SqlTools.ServiceLayer.FileBrowser.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.FileBrowser
+{
+    /// <summary>
+    /// Implementation for file browser operation
+    /// </summary>
+    internal class FileBrowserOperation : FileBrowserBase
+    {
+        private FileTree fileTree;
+        private string expandPath;
+        private string[] fileFilters;
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileBrowser"/> class.
+        /// </summary>
+        public FileBrowserOperation()
+        {
+            this.fileTree = new FileTree();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileBrowser"/> class.
+        /// </summary>
+        /// <param name="connectionInfo">The connection info</param>
+        /// <param name="fileFilters">The file extension filters</param>
+        public FileBrowserOperation(object connectionInfo, string expandPath, string[] fileFilters = null): this()
+        {
+            this.sqlConnection = connectionInfo;
+            this.expandPath = expandPath;
+            if (fileFilters == null)
+            {
+                this.fileFilters = new string[1] { "*" };
+            }
+            else
+            {
+                this.fileFilters = fileFilters;
+            }
+        }
+
+        #endregion
+
+        #region public properties and methods
+
+        public FileTree FileTree
+        {
+            get
+            {
+                return this.fileTree;
+            }
+        }
+
+        internal string[] FileFilters
+        {
+            get
+            {
+                return this.fileFilters;
+            }
+        }
+
+        public void PopulateFileTree()
+        {
+            this.PathSeparator = GetPathSeparator(this.Enumerator, this.sqlConnection);
+            PopulateDrives();
+            ExpandSelectedNode(this.expandPath);
+        }
+
+        /// <summary>
+        /// Expand nodes for the selected path.
+        /// </summary>
+        public void ExpandSelectedNode(string expandPath)
+        {
+            this.expandPath = expandPath;
+            if (!String.IsNullOrEmpty(this.expandPath))
```

Comment:
>String [](start = 17, length = 6)

please use string instead of String for call static methods on primitive types per the coding convention of this repo.